### PR TITLE
Add "PostRendererArgs" option to be passed to helm

### PR DESCRIPTION
**Pull request by boekhold**
This allows using PowerShell scripts on Windows as Post Renderer.

I have tried to follow the same style everywhere as the code for "--post-renderer". This has been tested on an actual complex chart on Windows 10.

I have not added any test cases, as that is currently beyond my golang skills (today was literally the first time I wrote something in Go).

### CODE REVIEW

**Comment by yxxhero** [comment]
@boekhold please fix DCO issue.

**Comment by boekhold** [comment]
> @boekhold please fix DCO issue.

Done, I think...

**Comment by yxxhero** [comment]
@boekhold please fix ci issue.

**Comment by boekhold** [comment]
> @boekhold please fix ci issue.

I've fixed the reported Linter issue. If I run golangci-lint locally, I now do not get any errors. I'm wondering however: there are 2 other places in the code I added that first check for `x != nil` before calling `len(x) != 0`, and the linter does not report them. I'll tag comments on those lines. Should I remove the `x != nil` check from those lines as well?

About the failing test cases, the failing one is TestGenerateID. I think what is happening is that since I expanded the ReleaseSpec struct with a new element, I somehow changed the hash that is generated for the ReleaseSpec with test data? Am I supposed to update `temp_test.go` with the new hashes as reported by `got (+)` ? Appreciate some guidance here.

File: pkg/state/helmx.go [hunk 1]
```diff
@@ -39,6 +39,31 @@ func (st *HelmState) appendPostRenderFlags(flags []string, release *ReleaseSpec,
...
+				flags = append(flags, "--post-renderer-args", arg)
+			}
+		}
+	case len(postRendererArgs) != 0:
+		for _, arg := range postRendererArgs {
+			if arg != "" {
+				flags = append(flags, "--post-renderer-args", arg)
+			}
+		}
+	case st.HelmDefaults.PostRendererArgs != nil && len(st.HelmDefaults.PostRendererArgs) != 0:
```
Comment (boekhold) [inline pkg/state/helmx.go:57]: Should I also remove the `x != nil` check here?

**Comment by boekhold** [comment]
> > @boekhold please fix ci issue.
> About the failing test cases, the failing one is TestGenerateID. I think what is happening is that since I expanded the ReleaseSpec struct with a new element, I somehow changed the hash that is generated for the ReleaseSpec with test data? Am I supposed to update `temp_test.go` with the new hashes as reported by `got (+)` ? Appreciate some guidance here.

Latest commit updates the expected hashes in `temp_test.go`. This should fix all 'ci' issues.

File: pkg/state/helmx.go [hunk 2]
```diff
@@ -39,6 +39,31 @@ func (st *HelmState) appendPostRenderFlags(flags []string, release *ReleaseSpec,
 	return flags
 }
 
+// append post-renderer-args flags to helm flags
+func (st *HelmState) appendPostRenderArgsFlags(flags []string, release *ReleaseSpec, postRendererArgs []string) []string {
+	switch {
+	case release.PostRendererArgs != nil && len(release.PostRendererArgs) != 0:
```
[the remark this answers is not in the document]
Comment (yxxhero) [reply pkg/state/helmx.go:45]: yeah. PostRendererArgs is [], len() judge is ok.

[the remark this answers is not in the document]
Comment (boekhold) [reply pkg/state/helmx.go]: Unnecessary nil checks removed

**Comment by yxxhero** [comment]
@boekhold could you add some extra e2e tests? and then I will merge this.

**Comment by boekhold** [comment]
@yxxhero I'll need a lot of time for that. As mentioned I am completely new to the Go ecosystem so will need a lot of studying on how to write one... I'm willing, just letting you know it will take time!

**Comment by yxxhero** [comment]
please ref: https://github.com/helmfile/helmfile/pull/1104 @boekhold

**Comment by boekhold** [comment]
> @boekhold could you add some extra e2e tests? and then I will merge this.

@yxxhero I honestly have no idea what I should do here. I had a look at the `tests/e2e` directory, and I cannot find any example test cases where CLI parameter behavior is being tested.

Can you provide some guidance/pointers to examples of what you would like me to add?

**Comment by yxxhero** [comment]
@boekhold I will help you to add tests.

**Comment by yxxhero** [comment]
@boekhold remove helmfile.exe

**Comment by yxxhero** [comment]
@boekhold  all is ok.

**Review by yxxhero — APPROVED** [review]

**Pull request merged**